src/odemis/driver/smaract.py

```diff
@@ -4218,20 +4225,29 @@ def __init__(self, name, role, locator, channels, ref_on_init=False, precision_m
         """
         model.HwComponent.__init__(self, name, role, *args, **kwargs)
 
+        if not smaract_python_sdk:
+            raise ModuleNotFoundError("Smaract Python SDK modules are missing. Cannot run Picoscale driver. "
+                                      "Install them by running: sudo apt install python3-smaract-si python3-smaract-picoscale.")
+
         # Connection
         if locator == "fake":
-            self.core = FakePicoscale_DLL()
+            self.core = FakePicoscale()
         else:
-            self.core = SA_SIDLL()
-        self._locator = locator
-        self._id = self._openConnection(locator)
+            self.core = si
```

Comment:
~I don't see `si` defined anywhere, should it be part of the keyword arguments?~

Reply:
I just noticed that it comes from:
```
    import smaract.si as si
    import smaract.picoscale as ps
```
In some places you use `si.` and in others `self.core.`, what's the difference?

Reply:
`self.core.` has the methods (from si module itself or from the simulator), while the use of `si.` is for constants